--- cinnamon-control-center-4.8.2/panels/region/cinnamon-region-panel-xkb.h.orig	2021-01-02 19:26:58.000000000 +0100
+++ cinnamon-control-center-4.8.2/panels/region/cinnamon-region-panel-xkb.h	2023-05-14 18:01:28.027956489 +0200
@@ -36,6 +36,12 @@ extern GSettings *xkb_keyboard_settings;
 extern GSettings *xkb_desktop_settings;
 extern GkbdKeyboardConfig initial_config;
 
+/* removed from libgnomekbd 3.28+ API */
+extern const gchar GKBD_DESKTOP_CONFIG_KEY_DEFAULT_GROUP[];
+extern const gchar GKBD_DESKTOP_CONFIG_KEY_GROUP_PER_WINDOW[];
+extern const gchar GKBD_KEYBOARD_CONFIG_KEY_LAYOUTS[];
+extern const gchar GKBD_KEYBOARD_CONFIG_KEY_OPTIONS[];
+
 extern void setup_xkb_tabs (GtkBuilder * dialog);
 
 extern void xkb_layouts_fill_selected_tree (GtkBuilder * dialog);
--- cinnamon-control-center-4.8.2/panels/region/cinnamon-region-panel-xkb.c.orig	2021-01-02 19:26:58.000000000 +0100
+++ cinnamon-control-center-4.8.2/panels/region/cinnamon-region-panel-xkb.c	2023-05-14 18:02:02.957767258 +0200
@@ -36,6 +36,12 @@
 
 #define GKBD_CONFIG_KEY_LOAD_EXTRA_ITEMS "load-extra-items"
 
+/* removed from libgnomekbd 3.28+ API */
+const gchar GKBD_DESKTOP_CONFIG_KEY_DEFAULT_GROUP[] = "default-group";
+const gchar GKBD_DESKTOP_CONFIG_KEY_GROUP_PER_WINDOW[] = "group-per-window";
+const gchar GKBD_KEYBOARD_CONFIG_KEY_LAYOUTS[] = "layouts";
+const gchar GKBD_KEYBOARD_CONFIG_KEY_OPTIONS[] = "options";
+
 XklEngine *engine;
 XklConfigRegistry *config_registry;
 
